src/api/ledger/transaction.js

```diff
@@ -1,28 +1,22 @@
 /* @flow */
 'use strict';
 const _ = require('lodash');
-const async = require('async');
 const utils = require('./utils');
 const parseTransaction = require('./parse/transaction');
-const {validate, convertErrors, errors} = utils.common;
-const RippleError = require('../../core/rippleerror').RippleError;
+const {validate, errors} = utils.common;
+import type {Connection} from '../common/connection.js';
+import type {TransactionType, TransactionOptions} from './transaction-types';
 
-import type {Remote} from '../../core/remote';
-
-import type {CallbackType, TransactionType,
-  GetTransactionResponseCallback, TransactionOptions}
-  from './transaction-types';
-
-function attachTransactionDate(remote: Remote, tx: Object,
-                              callback: CallbackType
-) {
+function attachTransactionDate(connection: Connection, tx: Object
+): Promise<TransactionType> {
   if (tx.date) {
-    callback(null, tx);
-    return;
+    return new Promise(resolve => resolve(tx));
```

Comment:
Ah, I was looking for that, not sure why I didn't see it in the docs. I'll make that change.